### ISSUE TITLE
Gson integration test fails with Camel 3.1.0 

### ISSUE THREAD

**Issue by jamesnetherton** [apache/camel-quarkus#776]
This slipped through the Camel 3.1.0 upgrade due to #723.

```
2020-02-27 16:54:43,873 ERROR [org.apa.cam.imp.eng.AbstractCamelContext] (main) Error starting CamelContext (camel-quarkus-integration-tests-dataformats-json) due to exception thrown: Failed to create route route16 at: >>> Unmarshal[org.apache.camel.model.dataformat.JsonDataFormat@58ca6ba3] <<< in route: Route(route16)[From[direct:Gson-type-as-header] -> [SetHeade... because of Error binding property (allowUnmarshallType=true) with name: allowUnmarshallType on bean: org.apache.camel.component.gson.GsonDataFormat@52fe87e0 with value: true: org.apache.camel.FailedToCreateRouteException: Failed to create route route16 at: >>> Unmarshal[org.apache.camel.model.dataformat.JsonDataFormat@58ca6ba3] <<< in route: Route(route16)[From[direct:Gson-type-as-header] -> [SetHeade... because of Error binding property (allowUnmarshallType=true) with name: allowUnmarshallType on bean: org.apache.camel.component.gson.GsonDataFormat@52fe87e0 with value: true
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:390)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:112)
	at org.apache.camel.quarkus.core.FastModel.start(FastModel.java:57)
	at org.apache.camel.quarkus.core.BaseModel.startRoute(BaseModel.java:327)
	at org.apache.camel.quarkus.core.BaseModel.startRouteDefinitions(BaseModel.java:320)
	at org.apache.camel.quarkus.core.BaseModel.startRouteDefinitions(BaseModel.java:299)
	at org.apache.camel.quarkus.core.FastCamelContext.startRouteDefinitions(FastCamelContext.java:141)
	at org.apache.camel.impl.engine.AbstractCamelContext.doStartCamel(AbstractCamelContext.java:2679)
	at org.apache.camel.impl.engine.AbstractCamelContext.lambda$doStart$2(AbstractCamelContext.java:2527)
	at org.apache.camel.impl.engine.AbstractCamelContext.doWithDefinedClassLoader(AbstractCamelContext.java:2544)
	at org.apache.camel.impl.engine.AbstractCamelContext.doStart(AbstractCamelContext.java:2525)
	at org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:121)
	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2421)
	at org.apache.camel.quarkus.core.CamelMain.doStart(CamelMain.java:50)
	at org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:121)
	at org.apache.camel.quarkus.core.CamelMainRecorder.start(CamelMainRecorder.java:97)
	at io.quarkus.deployment.steps.Main$start41.deploy_0(Main$start41.zig:78)
	at io.quarkus.deployment.steps.Main$start41.deploy(Main$start41.zig:97)
	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:218)
	at io.quarkus.runtime.Application.start(Application.java:89)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:95)
	at io.quarkus.test.junit.QuarkusTestExtension.doJavaStart(QuarkusTestExtension.java:88)
	at io.quarkus.test.junit.QuarkusTestExtension.ensureStarted(QuarkusTestExtension.java:171)
	at io.quarkus.test.junit.QuarkusTestExtension.beforeAll(QuarkusTestExtension.java:199)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:154)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:127)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
Caused by: org.apache.camel.PropertyBindingException: Error binding property (allowUnmarshallType=true) with name: allowUnmarshallType on bean: org.apache.camel.component.gson.GsonDataFormat@52fe87e0 with value: true
	at org.apache.camel.support.PropertyBindingSupport.lambda$doBindProperties$1(PropertyBindingSupport.java:530)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.util.stream.SortedOps$SizedRefSortingSink.end(SortedOps.java:357)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:483)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:485)
	at org.apache.camel.support.PropertyBindingSupport.doBindProperties(PropertyBindingSupport.java:513)
	at org.apache.camel.support.PropertyBindingSupport.access$000(PropertyBindingSupport.java:61)
	at org.apache.camel.support.PropertyBindingSupport$Builder.bind(PropertyBindingSupport.java:227)
	at org.apache.camel.reifier.dataformat.DataFormatReifier.configureDataFormat(DataFormatReifier.java:252)
	at org.apache.camel.reifier.dataformat.DataFormatReifier.createDataFormat(DataFormatReifier.java:209)
	at org.apache.camel.reifier.dataformat.DataFormatReifier.getDataFormat(DataFormatReifier.java:187)
	at org.apache.camel.reifier.UnmarshalReifier.createProcessor(UnmarshalReifier.java:35)
	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:787)
	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:513)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:388)
	... 64 more

```

**Comment by aldettinger** [comment]
Thanks for spotting, I'll take a look as it may impact #775 anyway.

**Comment by aldettinger** [comment]
I'm gonna fake the test until [CAMEL-14630](https://issues.apache.org/jira/browse/CAMEL-14630) has been solved, hopefully in camel 3.2.

Then we could remove the fake direct:Gson-type-as-header xml route, and put back the original test route, like:
        setHeader name="CamelGsonUnmarshalType">
            <constant>org.apache.camel.quarkus.component.jackson.model.DummyObject</constant>
        </setHeader>
        <unmarshal>
            <json library="Gson" allowUnmarshallType="true"/>
        </unmarshal>
